### PR TITLE
Chore

### PR DESCRIPTION
**Pull request by kurumeii**
- Use function to create schema for better locale support
- Update axios config
- Redirect to login if no access token
- Update login api
- Update type

### CODE REVIEW

File: src/config/schema/index.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import { capitalize } from 'lodash-es'
+import { type TFunction } from 'next-i18next'
+import { z } from 'zod'
+
+export const createLoginSchema = <T extends TFunction>(t: T) => {
+  return z.object({
+    usernameOrEmail: z.string().min(1, {
+      message: capitalize(t('auth:validation.usernameOrEmail.isRequired')),
+    }),
+    password: z
```
Comment (hoanhxlyn) [inline src/config/schema/index.ts:11]: a lot of duplicated code logic, should abstract it

File: src/config/lib/axiosConfig.ts [hunk 2]
```diff
@@ -10,13 +10,15 @@ const axiosClient = axios.create({
   baseURL: env.NEXT_PUBLIC_API_URL,
   headers: {
     Accept: 'application/json',
-    'Content-Type': 'application/json',
+    'Content-Type': 'application/json; charset=utf-8',
```
[the remark this answers is not in the document]
Comment (kurumeii) [reply src/config/lib/axiosConfig.ts:13]: no but it makes things clearer

File: src/config/schema/index.ts (hunk 1 above)
Comment (kurumeii) [reply src/config/schema/index.ts:11]: It will defeat the purpose of using helper functions instead of normal zod schema in the first place

Comment (hoanhxlyn) [reply src/config/schema/index.ts:11]: only abstract the most used schema such as email and password. Like so
```jsx
password: createPasswordSchema(t)
```

Comment (kurumeii) [reply src/config/schema/index.ts:11]: Sounds fair, I will try in the next PR

**Review by hoanhxlyn — APPROVED** [review]

**Pull request merged**